packages/eslint-config-typescript/index.js

```diff
@@ -1,226 +1,233 @@
-            { accessibility: 'explicit', overrides: { constructors: 'off' } }
-        ],
-
-        /*
-            All interface members should be terminated with a semicolon including single line
-            definitions, consistent with classes. Object literal types should use commas
-            consistent with object literals.
-        */
-        '@typescript-eslint/member-delimiter-style': ['error', {
-            overrides: {
-                interface: {
-                    singleline: {
-                        delimiter: 'semi',
-                        requireLast: true
-                    }
-                },
-                typeLiteral: {
-                    multiline: {
-                        delimiter: 'comma',
-                        requireLast: false
+import javascript from '@ni/eslint-config-javascript';
+import typescriptPlugin from 'typescript-eslint';
+import importPlugin from 'eslint-plugin-import';
+import extensions from './lib/extensions.js';
+
+export default [
+    ...javascript,
+    ...typescriptPlugin.configs.recommended,
+    ...extensions,
+    {
+        languageOptions: {
+            parser: typescriptPlugin.parser
+        },
+        settings: importPlugin.flatConfigs.typescript.settings,
+        rules: {
+            /*
+                Overrides to import rules (already handled by the TypeScript compiler):
+                https://github.com/benmosher/eslint-plugin-import/blob/main/config/typescript.js
+            */
+
+            /*
+                Already enforced by TypeScript compiler.
+            */
+            'import/named': 'off',
+
+            /*
+                Already enforced by TypeScript compiler.
+            */
+            'import/no-unresolved': 'off',
+
+            /*
+                Overrides to TypeScript recommended rules:
+                https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
+            */
+
+            '@typescript-eslint/explicit-module-boundary-types': 'error',
+
+            /*
+                [strict-null-checks]
+                With `stickNullChecks` disabled, non-null assertions are discouraged (this rule is configured as 'error')
+                as non-null assertions do not provide additional type safety and if used inconsistently add to visual noise.
+                With `strictNullChecks` enabled, the non-null assertion operator (this rule configured as 'off')
+                is useful for asserting that a value will no longer be null and usage of operator should be allowed.
+            */
+            '@typescript-eslint/no-non-null-assertion': 'error',
+
+            'valid-typeof': 'off',
+
+            /*
+                Overrides to TypeScript stylistic rules:
+                https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/stylistic.ts
+            */
+
```

Comment:
I didn't notice this in the previous PR but it's weird to have an empty section. Could you either delete this comment or change it to say that there aren't any overrides?

[Edit]Actually, it looks like there are some stylistic overrides below. Can you move this down there?